core/notification/router_receiver_service.go

```diff
@@ -26,10 +28,26 @@ func (s *RouterReceiverService) PrepareMetaMessages(ctx context.Context, n Notif
 		return nil, nil, errors.ErrInvalid.WithMsgf("number of receiver selectors should be less than or equal threshold %d", s.deps.Cfg.MaxNumReceiverSelectors)
 	}
 
-	rcvs, err := s.deps.ReceiverService.List(ctx, receiver.Filter{
-		MultipleLabels: n.ReceiverSelectors,
-		Expanded:       true,
-	})
+	var rcvs []receiver.Receiver
+	var userConfigs map[uint64]map[string]interface{}
+
+	// Check if any selector contains a config
+	hasConfig := false
+	for _, selector := range n.ReceiverSelectors {
+		if _, ok := selector["config"]; ok {
+			hasConfig = true
+			break
+		}
+	}
+
+	if hasConfig {
+		// Handle case when config is provided
+		rcvs, userConfigs, err = s.handleConfigCase(ctx, n.ReceiverSelectors)
+	} else {
+		// Handle case when only receiver IDs are provided
+		rcvs, err = s.handleIDOnlyCase(ctx, n.ReceiverSelectors)
+	}
```

Comment:
why only receiver IDs?

Reply:
Naming is not correct it is handling all the cases when config is not passed in receivers.